File: Carly_Michael/jectacular-ql/src/app/domain/ast/ql/expressions/unary-expression.ts
import {Expression} from './expression';
import {Location} from '../../location';
import {ExpressionVisitor} from '../visitors/expression-visitor';

export abstract class UnaryExpression extends Expression {
  constructor(public readonly right: Expression, readonly location: Location) {
    super(location);
  }
}

export class NegativeExpression extends UnaryExpression {
  constructor(readonly right: Expression, readonly location: Location) {
    super(right, location);
  }

  accept<T>(visitor: ExpressionVisitor<T>): T {
    return visitor.visitNegativeExpression(this);
  }
}

export class NegateExpression extends UnaryExpression {
  constructor(readonly right: Expression, readonly location: Location) {
    super(right, location);
  }

  accept<T>(visitor: ExpressionVisitor<T>): T {
    return visitor.visitNegateExpression(this);
  }
}

File: Remi_Hasan/src/main/java/ql/model/expression/unary/UnaryExpression.java
package ql.model.expression.unary;

import ql.model.expression.Expression;

public abstract class UnaryExpression extends Expression {
    private final Expression operand;

    UnaryExpression(Expression operand) {
        this.operand = operand;
    }

    public Expression getOperand() {
        return operand;
    }
}